src/js/gi/containers/SearchPage.jsx

```diff
@@ -10,22 +18,97 @@ import Pagination from '../../common/components/Pagination';
 
 export class SearchPage extends React.Component {
 
-  componentWillMount() {
-    this.props.fetch();
+  constructor(props) {
+    super(props);
+    this.handlePageSelect = this.handlePageSelect.bind(this);
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.updateSearchResults = this.updateSearchResults.bind(this);
   }
 
   componentDidMount() {
     let title = 'Search Results';
     const searchTerm = this.props.autocomplete.term;
     if (searchTerm) { title += ` - ${searchTerm}`; }
     this.props.setPageTitle(title);
+    this.updateSearchResults();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(this.props.location, prevProps.location)) {
+      this.updateSearchResults();
+    }
+  }
+
+  updateSearchResults() {
+    const programFilters = [
+      'caution',
+      'studentVeteranGroup',
+      'yellowRibbonScholarship',
+      'principlesOfExcellence',
+      'eightKeysToVeteranSuccess'
+    ];
+
+    const query = _.pick(this.props.location.query, [
+      'page',
+      'name',
+      'country',
+      'state',
+      'typeName',
+      ...programFilters
+    ]);
+
+    // Update form selections based on query.
+    const institutionFilter = _.omit(query, ['page', 'name']);
+
+    // Convert string to bool for params associated with checkboxes.
+    programFilters.forEach(filterKey => {
+      const filterValue = institutionFilter[filterKey];
+      institutionFilter[filterKey] =
+        filterValue === 'true' ||
+        (filterKey === 'caution' && filterValue === 'false');
+    });
+
+    // Derive type selection from typeName in the query.
+    if (institutionFilter.typeName) {
+      const typeName = institutionFilter.typeName.toLowerCase();
+      if (typeName === 'school' || typeName === 'employer') {
+        institutionFilter.type = typeName;
+      }
+    }
+
+    this.props.institutionFilterChange(institutionFilter);
+    this.props.fetchSearchResults(query);
   }
 
   handlePageSelect(page) {
-    this.props.fetch(page);
+    this.props.router.push({
+      ...this.props.location,
+      query: { ...this.props.location.query, page }
+    });
+  }
+
+  handleFilterChange(field, value) {
+    // Translate form selections to query params.
+    const queryKey = field === 'type' ? 'typeName' : field;
+    const queryValue = field === 'caution' ? !value : value;
+    const query = { ...this.props.location.query, [queryKey]: queryValue };
+
+    const shouldRemoveFilter =
+      (queryKey !== 'caution' && !queryValue) ||
+      (queryKey === 'caution' && queryValue) ||
+      ((queryKey === 'country' ||
+        queryKey === 'state' ||
+        queryKey === 'typeName') && queryValue === 'ALL');
+
+    if (shouldRemoveFilter) { delete query[queryKey]; }
+    this.props.router.push({ ...this.props.location, query });
```

Comment:
In this handler, form input events trigger a URL query update, which invokes `updateSearchResults` from `componentDidUpdate` to actually update the state of the form and also `fetchSearchResults`.